app/controllers/concerns/alchemy/admin/current_language.rb

```diff
@@ -13,7 +13,11 @@ module CurrentLanguage
 
       def load_current_language
         @current_language = if session[:alchemy_language_id].present?
-          set_alchemy_language(session[:alchemy_language_id])
+          if Current.site.languages.where(id: session[:alchemy_language_id]).exists?
```

Comment:
Couldn't we just clear the `language_id` in the `session` whenever we switch a site? I would rather not have to make this check every time we load the language (what we do on every request on most Admin screens)

Reply:
@tvdeyen that sounds like a good idea.  However, I'm having a hard time finding where `Alchemy::Current.site` is set when on `admin` actions.  I was assuming I could make the change wherever that happens based on the change to the `site_id` param.  Could you point me in the right direction?

Reply:
It happens in the `Admin::BaseController`

https://github.com/AlchemyCMS/alchemy_cms/blob/2dd59e87f360660b7d78817689ea20da9ab0d483/app/controllers/alchemy/admin/base_controller.rb#L139-L146

We would need to only reset the language_id if we have a site_id param